solr/core/src/test/org/apache/solr/handler/admin/MetricsHandlerTest.java

```diff
@@ -677,6 +706,340 @@ public void testExprMetrics() throws Exception {
+    // availableProcessors. We confirm snapshot exists and is nonNull instead.
+    MetricsHandler handler = new MetricsHandler(h.getCoreContainer());
+
+    SolrQueryResponse resp = new SolrQueryResponse();
+    handler.handleRequestBody(
+        req(
+            CommonParams.QT,
+            "/admin/metrics",
+            MetricsHandler.COMPACT_PARAM,
+            "false",
+            CommonParams.WT,
+            "prometheus"),
+        resp);
+
+    NamedList<?> values = resp.getValues();
+    assertNotNull(values.get("metrics"));
+    values = (NamedList<?>) values.get("metrics");
+    SolrPrometheusExporter exporter = (SolrPrometheusExporter) values.get("solr.jvm");
+    assertNotNull(exporter);
+    MetricSnapshots actualSnapshots = exporter.collect();
+    assertNotNull(actualSnapshots);
+
+    MetricSnapshot actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_gc");
+    GaugeSnapshot.GaugeDataPointSnapshot actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(actualSnapshot, Labels.of("item", "G1-Old-Generation"));
+    assertEquals(0, actualGaugeDataPoint.getValue(), 0);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_threads");
+    actualGaugeDataPoint = getGaugeDatapointSnapshot(actualSnapshot, Labels.of("item", "count"));
+    assertNotNull(actualGaugeDataPoint);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_memory_pools_bytes");
+    actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(
+            actualSnapshot, Labels.of("item", "committed", "space", "CodeHeap-'non-nmethods'"));
+    assertEquals(0, actualGaugeDataPoint.getValue(), 0);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_os");
+    actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(actualSnapshot, Labels.of("item", "availableProcessors"));
+    assertNotNull(actualGaugeDataPoint);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_buffers");
+    actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(actualSnapshot, Labels.of("item", "Count", "pool", "direct"));
+    assertNotNull(actualGaugeDataPoint);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_heap");
+    actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(actualSnapshot, Labels.of("item", "committed", "memory", "heap"));
+    assertNotNull(actualGaugeDataPoint);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_buffers_bytes");
+    actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(
+            actualSnapshot, Labels.of("item", "MemoryUsed", "pool", "direct"));
+    assertNotNull(actualGaugeDataPoint);
+
+    actualSnapshot = getMetricSnapshot(actualSnapshots, "solr_metrics_jvm_gc_seconds");
+    actualGaugeDataPoint =
+        getGaugeDatapointSnapshot(actualSnapshot, Labels.of("item", "G1-Old-Generation"));
```

Comment:
This assertion will fail for varying JVMs; skip this one.  Some others too above... it's okay to make the test looser / less specific.

Reply:
Removed a few more tests from the JVM registry